### PR TITLE
feat: implement address input

### LINKED ISSUE #17189: Implement Address input component
Implement this component in the quo2 component library as per the design spec.

Reuse Quo2 components wherever applicable.

Designs
https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=24917-226306&mode=design&t=YoWxH2odVGTV9z52-4

Example where this component is used:
https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=689-193000&mode=design&t=gM0fq5vp48hLFjwU-4

Acceptance Criteria:

component spec tests - https://github.com/status-im/status-mobile/blob/develop/doc/component-tests-overview.md
quo2 preview screen
export from quo2/core.cljs
following best practices - https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md

Note
Please align with design team before starting this

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #17189 

### Summary

Implement Address input

#### Platforms

- Android
- iOS

### Steps to test

- Open Status
- Open Quo2 previes
- Select Inputs > Address input
- Verify correct design and behavior of the component

status: ready

### CODE REVIEW

File: src/quo2/components/inputs/address_input/view.cljs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+            [react-native.clipboard :as clipboard]
+            [quo2.theme :as quo.theme]
+            [quo2.foundations.colors :as colors]
+            [quo2.components.icon :as icon]
+            [quo2.components.buttons.button.view :as button]
+            [quo2.components.inputs.address-input.style :as style]
+            [utils.i18n :as i18n]
+            [reagent.core :as reagent]))
+
+(def ^:private ens-regex #"^(?=.{5,255}$)([a-zA-Z0-9-]+\.)*[a-zA-Z0-9-]+\.[a-zA-Z]{2,}$")
```
Comment (J-Son89) [inline src/quo2/components/inputs/address_input/view.cljs]: should this be defined somewhere more common? 🤔

Comment (J-Son89) [reply src/quo2/components/inputs/address_input/view.cljs]: could also be passed in as a prop if we need different regex dependent on networks/ test networks 🤔

File: src/quo2/components/inputs/address_input/view.cljs [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+(defn- icon-color
+  [blur? override-theme]
```
Comment (J-Son89) [inline src/quo2/components/inputs/address_input/view.cljs]: we don't want to have the concept of an `override-theme`, whatever theme value is in the component is just the `theme`

File: src/quo2/core_spec.cljs [hunk 3]
```diff
@@ -28,6 +28,7 @@
     [quo2.components.graph.wallet-graph.component-spec]
     [quo2.components.inputs.input.component-spec]
     [quo2.components.inputs.profile-input.component-spec]
```
Comment (J-Son89) [inline src/quo2/core_spec.cljs]: nit: could we alpha sort the .inputs. namespaces here :)

File: src/quo2/components/inputs/address_input/style.cljs [hunk 4]
```diff
@@ -0,0 +1,38 @@
+(ns quo2.components.inputs.address-input.style
+  (:require [quo2.foundations.colors :as colors]
+            [quo2.components.markdown.text :as text]))
+
+(def container
+  {:padding-horizontal 20
+   :padding-top        8
+   :padding-bottom     16
+   :height             48
+   :width              "100%"
```
Comment (OmarBasem) [inline src/quo2/components/inputs/address_input/style.cljs]: is this needed?

**Review by OmarBasem — APPROVED** [review]

**Review by ajayesivan — APPROVED** [review]

**Review by jo-mut — APPROVED** [review]

File: src/quo2/components/inputs/address_input/view.cljs (hunk 2 above)
Comment (briansztamfater) [reply src/quo2/components/inputs/address_input/view.cljs]: Yeah, you are right, I changed it to `theme`

File: src/quo2/components/inputs/address_input/view.cljs (hunk 1 above)
Comment (briansztamfater) [reply src/quo2/components/inputs/address_input/view.cljs]: Moved to `status2.constants` namespace and pass it to the component via props

File: src/quo2/core_spec.cljs (hunk 3 above)
Comment (briansztamfater) [reply src/quo2/core_spec.cljs]: Sorted 😃

File: src/quo2/components/inputs/address_input/style.cljs (hunk 4 above)
Comment (briansztamfater) [reply src/quo2/components/inputs/address_input/style.cljs]: Not needed indeed, I removed it 👍

**Comment by briansztamfater** [comment]
@Francesca-G can you please review this PR? 🙏 ❤️

File: src/quo2/components/inputs/address_input/view.cljs [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+  [rn/view {:style style/clear-icon-container}
+   [icon/icon :i/loading
+    {:color (icon-color blur? theme)
+     :size  20}]])
+
+(defn- positive-state-icon
+  [theme]
+  [rn/view {:style style/clear-icon-container}
+   [icon/icon :i/positive-state
+    {:color (colors/theme-colors colors/success-50 colors/success-60 theme)
```
Comment (J-Son89) [inline src/quo2/components/inputs/address_input/view.cljs]: nit, imo we don't want to directly use colors/success-50(/60). Better imo to use:
(colors/theme-colors 
(colors/custom-color :success 50)
(colors/custom-color :success 60)
theme)
and we let the resolver mechanism handle the look up.

Comment (briansztamfater) [reply src/quo2/components/inputs/address_input/view.cljs]: Sure, I can change it, thanks for the suggestion. Doing a quick search I detected several places where we are making use of `colors/success-50` and `colors/success-60`, I don't have a strong opinion about it, but maybe we can define those colors as private so we can only access to them via `custom-color` fn and keys in `colors-map`.
Do we have any definition or discussion about this I could read about?

**Review by Francesca-G — COMMENTED** [review]
Here's the Figma [frame](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=3407%3A391873&mode=design&t=ILan5tMvvVhrBHqf-1) with the design review :)

File: src/quo2/components/inputs/address_input/view.cljs (hunk 5 above)
Comment (J-Son89) [reply src/quo2/components/inputs/address_input/view.cljs]: I don't think we have a discussion but we for sure can start one. I started a doc on colors in general. 
https://www.notion.so/handling-custom-colors-d4060bc87cfb4c268d19a77f0711d3a1

But yeah maybe this idea needs to be more formalised. 👍 I agree making them private would help. Similiar for network colors which @mmilad75 added to the custom-color resolve here. Network colors should be made private too so abstract it away imo: 
https://github.com/status-im/status-mobile/pull/17196/files#diff-2ad5069d9e72acebf9104695abaf82e7264cc4bfa13187904738f39003b61e9b

Comment (briansztamfater) [reply src/quo2/components/inputs/address_input/view.cljs]: Cool, I updated this implementation for now, we can follow up on another PR with a general refactor once we agree on these colors guidelines 💪

**Comment by briansztamfater** [comment]
@Francesca-G Unfortunately, I couldn't reproduce the cursor issue, I can see it normally when I focus on the input. Can you try again with latest build and also provide more info about the device you are using (device model, OS version, etc) or if you did some special steps to reproduce it? Thanks!

https://github.com/status-im/status-mobile/assets/18485527/19dfc232-8254-482d-9114-0effca4439ae

Also re-checked Paste button font and matches the Figma design. Let me know if there's something I did not see.



Adjusted clear button alignment 👍 

Let me know any additional info and thanks again for testing.

**Comment by Francesca-G** [comment]
@briansztamfater the issue with the missing cursor happened when opening the preview for the first time, without tapping anywhere, the component should've been in default style but it's already active. 
As you can see from my screen recording: 
- I open the address input component and the component is in "active" style missing the cursor
- I tap the component and the cursor appears, I'm able to type
- I tap elsewhere to stop typing, the component is in the correct "default" style 

Seems I'm not able to preview the "default" style without tapping on the component first.


https://github.com/status-im/status-mobile/assets/122543089/df58617e-267a-438f-9d02-01d0e5f5a1e0

**Comment by briansztamfater** [comment]
@Francesca-G now I see! Thanks for catching this issue 👁️ , should be fixed now!

File: src/quo2/components/inputs/address_input/style.cljs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            [quo2.components.markdown.text :as text]
+            [react-native.platform :as platform]))
+
+(def container
+  {:padding-horizontal 20
+   :padding-top        8
+   :padding-bottom     16
+   :height             48
+   :flex-direction     :row
+   :align-items        :flex-start})
```
Comment (J-Son89) [inline src/quo2/components/inputs/address_input/style.cljs:12]: nit: maybe it's good in advance to add a container-style prop here and merge it. i.e
```
(defn container [constainer-style]
  (merge {:padding-horizontal 20
   :padding-top        8
   :padding-bottom     16
   :height             48
   :flex-direction     :row
   :align-items        :flex-start} 
   container-style))
   ```

**Pull request merged**